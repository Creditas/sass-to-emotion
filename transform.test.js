const transform = require('./transform');

describe('transform', () => {
  it('simple class no vars', async () => {
    expect(
      await transform(
        `.button {
          display: inline-flex;
          align-items: center;
          justify-content: center;
        }`,
      ),
    ).toMatchSnapshot();
  });

  it('nested class no vars', async () => {
    expect(
      await transform(
        `.button {
          display: flex;
          align-items: center;
          justify-content: center;

          .foo {
            display: block;
          }
        }`,
      ),
    ).toMatchSnapshot();
  });

  it('sass vars to import', async () => {
    expect(
      await transform(
        `.button {
          color: $fe-brary-colour-primary-dark;

          .bar {
            font-size: $fe-brary-font-h6-font-size;
          }
        }`,
      ),
    ).toMatchSnapshot();
  });

  it('placeholder', async () => {
    expect(
      await transform(
        `// this comment would cause issues using the postcss parser vs postcss-scss
        %message-shared {
          border: 1px solid #ccc;
          padding: 10px;
          color: #333;
        }

        .message {
          @extend %message-shared;
        }

        .success {
          @extend %message-shared;
          border-color: green;
        }`,
      ),
    ).toMatchSnapshot();
  });

  it('mixins', async () => {
    expect(
      await transform(
        `@mixin ad-exact($width, $height) {
          width: $width;
          height: $height;
          color: $fe-brary-colour-primary-dark;
        }

        .adspot-468x60_728x90-pos-1-container {
          color: blue;
          @include ad-exact(125px, 700px);
        }`,
      ),
    ).toMatchSnapshot();
  });

  it('mixins nested', async () => {
    expect(
      await transform(
        `@mixin ad-exact($width, $height) {
          width: $width;
          height: $height;

          .bar {
            color: $fe-brary-colour-primary-dark;
            display: flex;
            align-items: center;

            .foo {
              display: black;
              align-items: start;
            }
          }

          display: block;
        }`,
      ),
    ).toMatchSnapshot();
  });
});
