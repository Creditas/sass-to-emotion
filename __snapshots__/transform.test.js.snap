// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform adds a FIXME comment 1`] = `
"import { css } from '@emotion/core';
import { listingDetailsAdSpacing } from '../variables';

export default css\`
  /* FIXME: Sass maths was detected in the line below, you must fix manually. */
  height: 90px + \${listingDetailsAdSpacing} * 2;
\`;
"
`;

exports[`transform const vars up top 1`] = `
"import { css } from '@emotion/core';

export const agentAvatarBorderExperimentColor = '#979797';

export const agentDetailsCarousel = css\`
  position: relative;
\`;
"
`;

exports[`transform custom include-media max-width 1`] = `
"import { css } from '@emotion/core';
import { media } from '@domain-group/fe-brary';

export default css\`
  display: none;

  \${media('<desktop')} {
    display: inline;
  }
\`;
"
`;

exports[`transform custom include-media min-width 1`] = `
"import { css } from '@emotion/core';
import { media } from '@domain-group/fe-brary';

export default css\`
  display: none;

  \${media('>=desktop')} {
    display: inline;
  }
\`;
"
`;

exports[`transform custom vars 1`] = `
"import { css } from '@emotion/core';
import { fooBarBazWhizz } from '../variables';

export default css\`
  display: none;
  color: \${fooBarBazWhizz};
\`;
"
`;

exports[`transform custom vars in include 1`] = `
"import { css } from '@emotion/core';
import { simpleArrow } from '../utils';
import { mapMarkerArrowSize } from '../variables';

export default css\`
            $map-marker-arrow-size: 8px;

            &::after {
              \${simpleArrow(
    customVars.mapMarkerArrowSize,
    customVars.mapMarkerArrowSize,
    'down',
    vars.colour.white,
  )}
              margin-left: \${mapMarkerArrowSize};
            }
          \`;
"
`;

exports[`transform default export for one only class 1`] = `
"import { css } from '@emotion/core';

export default css\`
  color: blue;
\`;
"
`;

exports[`transform default export for one only mixin 1`] = `
"import { css } from '@emotion/core';

export default function adExact(width, height) {
  return css\`
    width: \${width};
    height: \${height};
  \`;
}
"
`;

exports[`transform default export for one only placeholder 1`] = `
"import { css } from '@emotion/core';

export default css\`
  margin-bottom: 20px;
\`;
"
`;

exports[`transform does not export placeholder if its being used in the file 1`] = `
"import { css } from '@emotion/core';

const detailsMain = css\`
  padding: 18px 0;
  flex: 1 1 auto;
  min-width: 0;
  width: 100%;
\`;

export const foo = css\`
  \${detailsMain};
\`;
"
`;

exports[`transform exports placeholder if its not being used in the file 1`] = `
"import { css } from '@emotion/core';

export const detailsMain = css\`
  padding: 18px 0;
  flex: 1 1 auto;
  min-width: 0;
  width: 100%;
\`;

export const foo = css\`
  color: pink;
\`;
"
`;

exports[`transform handle mixins without args 1`] = `
"import { css } from '@emotion/core';

export const message = css\`
  \${resetTopMargin()}
  margin-bottom: 20px;
\`;

function resetTopMargin() {
  return css\`
    > *:first-child {
      margin-top: 0;
    }
  \`;
}
"
`;

exports[`transform handles multi value var refrences 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';

export default css\`
  border-bottom: 1px solid \${vars.colour.neutral300};
\`;
"
`;

exports[`transform imports fe-brary helpers 1`] = `
"import { css } from '@emotion/core';
import { buttonNormalize } from '@domain-group/fe-brary';

export default css\`
  \${buttonNormalize};
  position: absolute;
\`;
"
`;

exports[`transform imports fe-brary helpers with @extend 1`] = `
"import { css } from '@emotion/core';
import { variables as vars, buttonNormalize } from '@domain-group/fe-brary';

export const agentDetailsRightArrow = css\`
  \${buttonNormalize};
  color: \${vars.colour.primaryDark};
\`;

export const anotherRef = css\`
  \${buttonNormalize};
\`;
"
`;

exports[`transform imports fe-brary helpers with @include 1`] = `
"import { css } from '@emotion/core';
import { resetTopMargin } from '../utils';

export default css\`
  \${resetTopMargin()}
  position: absolute;
\`;
"
`;

exports[`transform imports non-fe-brary helpers and vars 1`] = `
"import { css } from '@emotion/core';
import { fePaListingDetailsWrapper } from '../utils';

export default css\`
  \${fePaListingDetailsWrapper};

  margin-top: -54px;
\`;
"
`;

exports[`transform imports non-fe-brary helpers with @include 1`] = `
"import { css } from '@emotion/core';
import { fooBar } from '../utils';

export default css\`
  \${fooBar()}
  position: absolute;
\`;
"
`;

exports[`transform mixin exported when not refrenced in file 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';

export default function adExact(width, height) {
  return css\`
    width: \${width};
    height: \${height};
    color: \${vars.colour.primaryDark};
  \`;
}
"
`;

exports[`transform mixin refrenced does not export 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';

function adExact(width, height) {
  return css\`
    width: \${width};
    height: \${height};
    color: \${vars.colour.primaryDark};
  \`;
}

export const adspot468X60728X90Pos1Container = css\`
  color: blue;
  \${adExact('125px', '700px')}
\`;
"
`;

exports[`transform mixins nested 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';

export default function adExact(width, height) {
  return css\`
    width: \${width};
    height: \${height};

    .bar {
      color: \${vars.colour.primaryDark};
      display: flex;
      align-items: center;

      .foo {
        display: black;
        align-items: start;
      }
    }

    display: block;
  \`;
}
"
`;

exports[`transform multiple nested 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';
import { fePaListingDetailsWrapper } from '../utils';

export const aboutDevelopment = css\`
  \${fePaListingDetailsWrapper};
  margin-bottom: 24px;
\`;

export const aboutDevelopmentLeft = css\`
  flex: 1;
  padding: 20px;
\`;

export const aboutDevelopmentName = css\`
  font-size: \${vars.font.baseFontSize};
\`;

export const aboutDevelopmentPolicy = css\`
  font-size: 44px;
\`;
"
`;

exports[`transform multiple vars formating 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';

export const agentNameSize = '$fe-brary-font-h5-font-size';
export const companyNameSize = '$fe-brary-font-h6-font-size';
export const agentAvatarBorderExperimentColor = '#979797';

export const agentDetailsCarousel = css\`
  position: relative;
\`;
"
`;

exports[`transform nested class no vars 1`] = `
"import { css } from '@emotion/core';

export const button = css\`
  display: flex;
  align-items: center;
  justify-content: center;
\`;

export const foo = css\`
  display: block;
\`;
"
`;

exports[`transform nested with state 1`] = `
"import { css } from '@emotion/core';

export const agentDetailsCarousel = css\`
  position: relative;
\`;

export const fooBarBaz = css\`
  &.bazzy {
    color: pink;
  }
\`;
"
`;

exports[`transform never print nested  1`] = `
"import { css } from '@emotion/core';
import { variables as vars, fontSmall, aTag } from '@domain-group/fe-brary';
import { fePaListingDetailsWrapper, fePaListingDetailsHeading } from '../utils';

export const aboutDevelopment = css\`
  \${fePaListingDetailsWrapper};
  margin-bottom: 24px;
\`;

export const aboutDevelopmentHeader = css\`
  \${fePaListingDetailsHeading};
  display: flex;
\`;

export const aboutDevelopmentContainer = css\`
  \${fontSmall};
  border: 1px solid \${vars.colour.neutral300};
  display: flex;
  position: relative;
\`;

export const aboutDevelopmentLink = css\`
  \${aTag};
  position: static;
  cursor: pointer;

  &::before {
    cursor: pointer;
    bottom: 0;
    left: 0;
  }
\`;
"
`;

exports[`transform never print nested media 1`] = `
"import { css } from '@emotion/core';
import { media } from '@domain-group/fe-brary';
import { fePaListingDetailsWrapper, fePaListingDetailsHeading } from '../utils';

export const additionalFeatures = css\`
  \${fePaListingDetailsWrapper};
  margin-bottom: 24px;

  \${media('>=mobile')} {
    width: 80%;
  }
\`;

export const additionalFeaturesHeader = css\`
  \${fePaListingDetailsHeading};
\`;

export const additionalFeaturesListing = css\`
  list-style-type: none;

  \${media('>=tablet')} {
    width: 33.33333%;
  }
\`;
"
`;

exports[`transform no new line breaks 1`] = `
"import { css } from '@emotion/core';
import { aTag } from '@domain-group/fe-brary';
import { fePaListingDetailsWrapper } from '../utils';

export const aboutDevelopment = css\`
  \${fePaListingDetailsWrapper};
  margin-bottom: 24px;
\`;

export const aboutDevelopmentLink = css\`
  \${aTag};
  position: static;
  cursor: pointer;
\`;
"
`;

exports[`transform non classname 1`] = `
"import { css } from '@emotion/core';

export const description = css\`
  margin-bottom: 24px;

  ::hover {
    color: pink;
  }

  p {
    margin-bottom: 55px;
  }
\`;

export const foo = css\`
  margin-bottom: 100px;
\`;
"
`;

exports[`transform pesudo elements/classes class nested in pseudo becomes a ref 1`] = `
"import { css } from '@emotion/core';

export const buttonIcon = css\`
  color: blue;
\`;

export const buttonCopy = css\`
            display: flex;
            cursor: pointer;

            &:hover {
              [class*='\${buttonIcon.name}'] {
                opacity: 1;
                transform: translateX(0);
              }
            }
          \`;
"
`;

exports[`transform pesudo elements/classes nested 1`] = `
"import { css } from '@emotion/core';

export const buttonCopy = css\`
  background: none;
  border: 0;

  &:focus,
  &:hover {
    outline: 0;
  }
\`;

export const domainIcon = css\`
  width: 14px;
  height: 14px;
\`;
"
`;

exports[`transform pesudo elements/classes nested pseudos are kept 1`] = `
"import { css } from '@emotion/core';

export default css\`
  margin-left: auto;

  &:not(:last-of-type) {
    padding-right: 12px;

    &::after {
      height: 18px;
      content: '';
    }
  }
\`;
"
`;

exports[`transform pesudo elements/classes top level pseudo classes 1`] = `
"import { css } from '@emotion/core';

export default css\`
  &:first-of-type {
    color: black;
  }
\`;
"
`;

exports[`transform placeholder 1`] = `
"import { css } from '@emotion/core';

const messageShared = css\`
  border: 1px solid #ccc;
  padding: 10px;
  color: #333;
\`;

export const message = css\`
  \${messageShared};
\`;

export const success = css\`
  \${messageShared};
  border-color: green;
\`;
"
`;

exports[`transform ref nested classes 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';

export const agentDetailsAgent = css\`
  color: blue;
\`;

export const agentDetailsRightArrow = css\`
          color: \${vars.colour.primaryDark};

          &.with-big-agents {
            [class*='\${agentDetailsAgent.name}'] {
              text-align: center;
            }
          }
        \`;
"
`;

exports[`transform removes pre double underscore text 1`] = `
"import { css } from '@emotion/core';

const detailsMain = css\`
  padding: 18px 0;
  flex: 1 1 auto;
  min-width: 0;
  width: 100%;
\`;

function detailsContainer() {
  return css\`
    color: black;
  \`;
}

export const foo = css\`
  \${detailsMain};
  \${detailsContainer()}
\`;
"
`;

exports[`transform sass vars to import 1`] = `
"import { css } from '@emotion/core';
import { variables as vars } from '@domain-group/fe-brary';

export const button = css\`
  color: \${vars.colour.primaryDark};
  display: flex;
\`;

export const bar = css\`
  font-size: \${vars.font.h6FontSize};
\`;
"
`;

exports[`transform simple class no vars 1`] = `
"import { css } from '@emotion/core';

export default css\`
  display: inline-flex;
  align-items: center;
  justify-content: center;
\`;
"
`;

exports[`transform top level vars 1`] = `
"export default '42px';
"
`;

exports[`transform top level vars ref 1`] = `
"import { css } from '@emotion/core';
import { externalVar } from '../variables';

const fooBarBaz = '42px';

export const foo = css\`
  height: \${fooBarBaz};
  width: \${externalVar};
\`;
"
`;

exports[`transform vanilla css media query 1`] = `
"import { css } from '@emotion/core';

export default css\`
  display: none;

  @media (min-width: 100px) {
    display: inline;
  }
\`;
"
`;

exports[`transform vars refrencing fe-brary variables 1`] = `
"import { variables as vars } from '@domain-group/fe-brary';

export const agentNameSize = '$fe-brary-font-h5-font-size';
export const agentBoxBoxShadowLeftRight = 'inset -1px 0 0 0 $agent-box-border-color-lighter, inset 1px 0 0 0 $agent-box-border-color-light';
"
`;
